File: con-modulos/Connect4.js
let { Game } = require('./model/Game');
let { GameView } = require('./view/GameView');
let { YesNoDialogView } = require('./view/YesNoDialogView');

class Connect4 {

  play() {
    const continueDialogView = new YesNoDialogView(`Do you want to continue? (yes/no)`);
    do {
      let game = new Game();
      new GameView(game).play();
      continueDialogView.read();
    } while (continueDialogView.isAffirmative());
  }
}

exports.Connect4 = Connect4;

File: designPatterns/prototype/src/main/es/connect4/Connect4.java
package main.es.connect4;

import main.es.connect4.models.Game;
import main.es.connect4.views.GameView;

public class Connect4 {

    protected GameView gameView;

    private Connect4() {
        this.gameView = new GameView(new Game());
    }

    private void play() {
        do {
            this.gameView.start();
            this.gameView.play();
        } while (this.gameView.resume());
    }

    public static void main(final String[] args) {
        new Connect4().play();
    }

}
